crates/eth-rpc/src/main.rs

```diff
@@ -15,12 +27,45 @@ async fn main() -> Result<()> {
     tracing_subscriber::FmtSubscriber::builder().with_env_filter(filter).finish().try_init()?;
 
     let starknet_config = StarknetConfig::from_env()?;
+
     let rpc_config = RPCConfig::from_env()?;
 
-    let provider = JsonRpcClientBuilder::with_http(&starknet_config).unwrap().build();
-    let kakarot_client = KakarotClient::new(starknet_config, provider);
+    let provider: StarknetProvider = match &starknet_config.network {
```

Comment:
`let starknet_provider: StarknetProvider = `

i find it difficult to follow what is what generally speaking, and that prepending with Eth / Starknet helps

Reply:
fixed